src/OrchardCore.Modules/OrchardCore.Forms/Assets/js/form-visibility-rules.js

```diff
@@ -0,0 +1,252 @@
+window.formVisibilityGroupRules = (function () {
+    function initialize(data) {
+
+        const inputElement = getInputByName(data.elementName);
+
+        if (!inputElement || data.action === 'None') {
+            return;
+        }
+
+        if (inputElement.type == 'checkbox' || inputElement.type == 'radio') {
+
+            inputElement.setAttribute('data-default-value', inputElement.checked ? 'on' : 'off');
+        } else {
+            inputElement.setAttribute('data-default-value', inputElement.value);
+        }
+
+        const widgetContainer = inputElement.closest('.widget');
+
+        processGroups(data, inputElement, widgetContainer, true);
+
+        triggerProperChangeEvent(inputElement);
+    }
+
+    function triggerProperChangeEvent(element) {
+
+        const tagName = element.tagName.toUpperCase();
```

Comment:
I think this is all upppercase by default. you don't need it,

```suggestion
        const tagName = element.tagName;
```